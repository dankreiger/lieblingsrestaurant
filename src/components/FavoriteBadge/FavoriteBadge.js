import React from 'react';
import { string } from 'prop-types';
import { FavoriteBadgeContainer } from './FavoriteBadge.styles';

const FavoriteBadge = ({ favoriteRanking }) => {
  return (
    <FavoriteBadgeContainer>
      <svg viewBox="0 0 36 36" className="badge">
        <circle cx="50%" cy="50%" r="50%" />
        <text x="50%" y="55%" textAnchor="middle" alignmentBaseline="middle">
          {favoriteRanking}
        </text>
      </svg>
    </FavoriteBadgeContainer>
  );
};

FavoriteBadge.propTypes = {
  favoriteRanking: string
};

export default FavoriteBadge;
