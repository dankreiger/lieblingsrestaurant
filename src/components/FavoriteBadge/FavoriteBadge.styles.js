import styled from 'styled-components';

export const FavoriteBadgeContainer = styled.div`
  position: absolute;
  left: 10px;
  top: 10px;
  .badge {
    width: 60px;
    fill: rgba(0, 0, 0, 0.4);
  }

  .badge text {
    font-size: 18px;
    fill: white;
  }
`;
