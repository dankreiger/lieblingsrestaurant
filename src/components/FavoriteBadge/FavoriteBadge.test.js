import React from 'react';
import FavoriteBadge from './FavoriteBadge';
import { shallow } from 'enzyme';

describe('FavoriteBadge', () => {
  let favoriteBadgeComponent;

  beforeEach(() => {
    favoriteBadgeComponent = shallow(<FavoriteBadge />);
  });
  describe('rendering', () => {
    test('renders as expected', () => {
      expect(favoriteBadgeComponent.length).toBe(1);
      expect(favoriteBadgeComponent).toMatchSnapshot();
    });
  });
});
