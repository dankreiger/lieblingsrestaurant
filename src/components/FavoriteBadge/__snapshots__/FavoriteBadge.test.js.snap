// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteBadge rendering renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoriteBadge />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <svg
        className="badge"
        viewBox="0 0 36 36"
      >
        <circle
          cx="50%"
          cy="50%"
          r="50%"
        />
        <text
          alignmentBaseline="middle"
          textAnchor="middle"
          x="50%"
          y="55%"
        />
      </svg>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <circle
            cx="50%"
            cy="50%"
            r="50%"
          />,
          <text
            alignmentBaseline="middle"
            textAnchor="middle"
            x="50%"
            y="55%"
          />,
        ],
        "className": "badge",
        "viewBox": "0 0 36 36",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cx": "50%",
            "cy": "50%",
            "r": "50%",
          },
          "ref": null,
          "rendered": null,
          "type": "circle",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alignmentBaseline": "middle",
            "children": undefined,
            "textAnchor": "middle",
            "x": "50%",
            "y": "55%",
          },
          "ref": null,
          "rendered": null,
          "type": "text",
        },
      ],
      "type": "svg",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  position: absolute;
  left: 10px;
  top: 10px;
  .badge {
    width: 60px;
    fill: rgba(0, 0, 0, 0.4);
  }

  .badge text {
    font-size: 18px;
    fill: white;
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <svg
          className="badge"
          viewBox="0 0 36 36"
        >
          <circle
            cx="50%"
            cy="50%"
            r="50%"
          />
          <text
            alignmentBaseline="middle"
            textAnchor="middle"
            x="50%"
            y="55%"
          />
        </svg>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <circle
              cx="50%"
              cy="50%"
              r="50%"
            />,
            <text
              alignmentBaseline="middle"
              textAnchor="middle"
              x="50%"
              y="55%"
            />,
          ],
          "className": "badge",
          "viewBox": "0 0 36 36",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cx": "50%",
              "cy": "50%",
              "r": "50%",
            },
            "ref": null,
            "rendered": null,
            "type": "circle",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alignmentBaseline": "middle",
              "children": undefined,
              "textAnchor": "middle",
              "x": "50%",
              "y": "55%",
            },
            "ref": null,
            "rendered": null,
            "type": "text",
          },
        ],
        "type": "svg",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  position: absolute;
  left: 10px;
  top: 10px;
  .badge {
    width: 60px;
    fill: rgba(0, 0, 0, 0.4);
  }

  .badge text {
    font-size: 18px;
    fill: white;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
